cypress/e2e/widgets/widget-pip-close.spec.ts

```diff
@@ -0,0 +1,204 @@
+}
+
+describe("Widget PIP", () => {
+    let synapse: SynapseInstance;
+    let user: UserCredentials;
+    let bot: MatrixClient;
+    let demoWidgetUrl: string;
+
+    function roomCreateAddWidgetPip(userRemove: 'leave' | 'kick' | 'ban') {
+        cy.createRoom({
+            name: ROOM_NAME,
+            invite: [bot.getUserId()],
+        }).then(roomId => {
+            // sets bot to Admin and user to Moderator
+            cy.getClient().then(matrixClient => {
+                return matrixClient.sendStateEvent(roomId, 'm.room.power_levels', {
+                    users: {
+                        [user.userId]: 50,
+                        [bot.getUserId()]: 100,
+                    },
+                });
+            }).as('powerLevelsChanged');
+
+            // bot joins the room
+            cy.botJoinRoom(bot, roomId).then(async () => {
+                // wait for bot to join the room, otherwise sometimes widget event is not found
+                // setup widget via state event
+                cy.getClient().then(async matrixClient => {
+                    const content: IWidget = {
+                        id: DEMO_WIDGET_ID,
+                        creatorUserId: 'somebody',
+                        type: DEMO_WIDGET_TYPE,
+                        name: DEMO_WIDGET_NAME,
+                        url: demoWidgetUrl,
+                    };
+                    await matrixClient.sendStateEvent(roomId, 'im.vector.modular.widgets', content, DEMO_WIDGET_ID);
+                });
+            }).as('botJoinedWidgetEventSent');
+
+            // open the room
+            cy.viewRoomByName(ROOM_NAME);
+
+            cy.all([
+                cy.get<string>("@powerLevelsChanged"),
+                cy.get<string>("@botJoinedWidgetEventSent"),
+            ]).then(() => {
+                cy.window().then(async win => {
+                    // wait for widget state event
+                    await waitForRoomWidget(win, DEMO_WIDGET_ID, roomId, true);
+
+                    // activate widget in pip mode
+                    win.mxActiveWidgetStore.setWidgetPersistence(DEMO_WIDGET_ID, roomId, true);
+
+                    // checks that pip window is opened
+                    cy.get(".mx_CallView_pip").should("exist");
+
+                    // checks that widget is opened in pip
+                    cy.accessIframe(`iframe[title="${DEMO_WIDGET_NAME}"]`).within({}, () => {
+                        cy.get("#demo").should('exist');
+                    }).then(async () => {
+                        const userId = user.userId;
+                        if (userRemove == 'leave') {
+                            cy.getClient().then(async matrixClient => {
+                                await matrixClient.leave(roomId);
+                            });
+                        } else if (userRemove == 'kick') {
+                            // wait, otherwise sometimes not work
+                            cy.wait(500).then(async () => {
```

Comment:
Some more info here would be great, nominally awaiting on some DOM state would be better

Reply:
Problem is that I don't have any good explanation of that. I could say that the issue was not there when this spec was committed. I think the issue appeared when 'matrix-js-sdk' was updated to something after '19.2.0-rc.1'. It looks strange to me that it happens with the 'kick' but never with 'ban' for example.

Reply:
Unfortunately it'll need some investigation, sharing the failure mode/error message would be helpful.

Reply:
without wait (or with wait 0) it fails on this line https://github.com/nordeck/matrix-react-sdk/blob/25e69ed5cdcc8fcf2e24babf01a20cf04092326b/cypress/e2e/widgets/widget-pip-close.spec.ts#L165 so the PIP dialog stays on the page and test fails.

here is how it looks:
<img width="518" alt="Screen Shot 2022-08-16 at 18 54 24" src="https://user-images.githubusercontent.com/9369306/184925514-b5fbf77e-6f24-4bd5-b4c7-5bad821b7f35.png">

what is strange to me is that "Bot removed Mike" is shown as a first line but should be somewhere below. There probably some race condition here.

Reply:
@t3chguy does it help or more information is needed? Do you think that this issue should be handled in this PR or maybe separately?

Reply:
Definitely as part of this PR, we have too many issues with flaky tests, and things like arbitrary wait times are bound to be flaky as CI performance fluctuates. Should you not just be asserting that the PIP closes? i.e `.should("not.exist");` or whatever

Reply:
@t3chguy it is asserted here: https://github.com/nordeck/matrix-react-sdk/blob/25e69ed5cdcc8fcf2e24babf01a20cf04092326b/cypress/e2e/widgets/widget-pip-close.spec.ts#L165 and fails

Reply:
Cypress assertions are retried for 10s, so a wait of half a second shouldn't impact anything, I cannot in good conscience accept a test which adds a delay for an unknown reason, I suggest trying to fix it or asking Travis for help when he returns as the original reviewer

Reply:
Hi @t3chguy, thanks for the feedback, the kick issue is fixed and there are no delays anymore, spec is fine in pipeline. 

Could you please do please review one more time? Thanks! :)